File: src/fractal/brain/_op/clone.py
"""

    *Clone*

  A clone function that also handles CPU/GPU transfers.

"""

from copy import deepcopy

from .._brain import Brain


class Clone(
    Brain,
):
    def clone(
        self,
        on_gpu: bool,
        test_mode: bool,
    ):
        network = deepcopy(self)
        if network.on_gpu():
            network.move_to_gpu()
        else:
            network.move_to_cpu()
        return network

File: src/fractal/brain/_op/clone.rs
//! A clone function that also handles CPU/GPU transfers.

pub trait Clone {
    fn clone(&self, on_gpu: bool, is_testing: bool);
}

impl Clone for Brain {
    fn clone(
        &self,
        on_gpu: bool,
        test_mode: bool,
    ):
        network = &self.clone()
        if network.on_gpu() {
            network.move_to_gpu()
        } else {
            network.move_to_cpu()
        }
        return network
}
